### ISSUE TITLE
Inconsistent attribute handling between netcdf4 and h5netcdf engines

### ISSUE THREAD

**Issue by rabernat** [pydata/xarray#5172]
I have found a netCDF file that cannot be decoded by xarray via the h5netcdf engine but CAN be decoded via netCDF4. This could be considered an h5netcdf bug, but I thought I would raise it first here for visibility.

This file will reproduce the bug
```
! wget 'https://esgf-world.s3.amazonaws.com/CMIP6/CMIP/IPSL/IPSL-CM6A-LR/abrupt-4xCO2/r1i1p1f1/Lmon/cLeaf/gr/v20190118/cLeaf_Lmon_IPSL-CM6A-LR_abrupt-4xCO2_r1i1p1f1_gr_185001-214912.nc'
```

```python
import netCDF4
import h5netcdf.legacyapi as netCDF4_h5

local_path = "cLeaf_Lmon_IPSL-CM6A-LR_abrupt-4xCO2_r1i1p1f1_gr_185001-214912.nc"
with netCDF4_h5.Dataset(local_path, mode='r') as ncfile:
    print('h5netcdf:', ncfile['cLeaf'].getncattr("coordinates"))
with netCDF4.Dataset(local_path, mode='r') as ncfile:
    #assert "coordinates" not in ncfile['cLeaf'].attrs
    print('netCDF4:', ncfile['cLeaf'].getncattr("coordinates"))
```

```
h5netcdf: Empty(dtype=dtype('S1'))
netCDF4: 
```

As we can see, we get an empty string `''` in netCDF4 but a `<class 'h5py._hl.base.Empty'>` object from h5netcdf. This weird attribute prevents xarray from decoding the dataset.

We could:
- Fix it in xarray, but having special handling for this sort of `Empty` object
- Fix it in h5netcdf

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27) 
[GCC 9.3.0]
python-bits: 64
OS: Linux
OS-release: 4.19.150+
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C.UTF-8
LANG: C.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.6
libnetcdf: 4.7.4

xarray: 0.17.0
pandas: 1.2.3
numpy: 1.20.2
scipy: 1.6.2
netCDF4: 1.5.6
pydap: installed
h5netcdf: 0.10.0
h5py: 3.1.0
Nio: None
zarr: 2.7.0
cftime: 1.4.1
nc_time_axis: 1.2.0
PseudoNetCDF: None
rasterio: 1.2.1
cfgrib: 0.9.8.5
iris: None
bottleneck: 1.3.2
dask: 2021.03.1
distributed: 2021.03.1
matplotlib: 3.3.4
cartopy: 0.18.0
seaborn: None
numbagg: None
pint: 0.17
setuptools: 49.6.0.post20210108
pip: 20.3.4
conda: None
pytest: None
IPython: 7.22.0
sphinx: None

</details>

xref https://github.com/pangeo-forge/pangeo-forge/issues/105

**Comment by kmuehlbauer** [comment]
@rabernat 

This seems to originate from `h5py`:

```python
import h5py
with h5py.File(local_path, mode="r", ) as ncfile:
    print(ncfile["cLeaf"].attrs["coordinates"])
```

```
Empty(dtype=dtype('S1'))
```

**Comment by rabernat** [comment]
Yes I agree. Should I just close this and move it to h5netcdf?

**Comment by kmuehlbauer** [comment]
@rabernat FYI, and if someone stumbles over this issue first. This is resolved with `h5netcdf = 0.11.0`.
